eras/babbage/impl/testlib/Test/Cardano/Ledger/Babbage/ImpTest.hs

```diff
@@ -37,7 +39,40 @@ instance ShelleyEraImp BabbageEra where
     defaultInitNewEpochState
       (nesEsL . curPParamsEpochStateL . ppCostModelsL <>~ testingCostModels [PlutusV2])
   impSatisfyNativeScript = impAllegraSatisfyNativeScript
-  fixupTx = alonzoFixupTx
+  fixupTx = babbageFixupTx
+
+babbageFixupTx ::
+  ( HasCallStack
+  , AlonzoEraImp era
+  , BabbageEraTxBody era
+  ) =>
+  Tx era ->
+  ImpTestM era (Tx era)
+babbageFixupTx =
+  addNativeScriptTxWits
+    >=> fixupAuxDataHash
+    >=> addCollateralInput
+    >=> addRootTxIn
+    >=> fixupScriptWits
+    >=> fixupOutputDatums
+    >=> fixupDatums
+    >=> fixupRedeemerIndices
+    >=> fixupTxOuts
+    >=> fixupCollateralReturn
+    >=> alonzoFixupFees
+    >=> fixupRedeemers
+    >=> fixupPPHash
+    >=> updateAddrTxWits
+
+fixupCollateralReturn ::
+  ( ShelleyEraImp era
+  , BabbageEraTxBody era
+  ) =>
+  Tx era ->
+  ImpTestM era (Tx era)
+fixupCollateralReturn tx = do
+  pp <- getsNES $ nesEsL . curPParamsEpochStateL
+  pure $ tx & bodyTxL . collateralReturnTxBodyL %~ fmap (setMinCoinTxOut pp)
```

Comment:
I suggest using `ensure` instead of `set`. This way if I want to use a higher than minimal value in the collateral return, I can do it.
```suggestion
  pure $ tx & bodyTxL . collateralReturnTxBodyL %~ fmap (ensureMinCoinTxOut pp)
```